test/update_handler.c

```diff
@@ -280,6 +280,13 @@ static void test_update_handler(UpdateHandlerFixture *fixture,
 			g_test_skip("RAUC_TEST_MTD_NAND undefined");
 			return;
 		}
+	} else if (g_strcmp0(test_pair->slottype, "flash") == 0) {
+		slotpath = g_strdup(g_getenv("RAUC_TEST_MTD_FLASH"));
```

Comment:
In the current qemu-test, `RAUC_TEST_MTD_FLASH` points to a nand-sim device, which is incorrect for this use-case (but will probably work, tough). Maybe this can be done with the mtdram driver and the correct module options. Or alternatively qemu might support emulating a NOR flash als on x86.

Reply:
>Maybe this can be done with the mtdram driver and the correct module options. Or alternatively qemu might support emulating a NOR flash als on x86.

I looked into NOR-flash emulation for qemu and this seems to be `difficult` so I'd rather use the mtdram driver.

Reply:
Currently `RAUC_TEST_MTD_FLASH` points to nowhere, the variable for nand flashes is `RAUC_TEST_MTD_NAND`.

Reply:
I adapted the qemu tests to run on the mtdram device. I also added `cat /proc/mtd` to the tests, this should make debugging easier if the device order changes for whatever reason.